File: Fase3/aws-rekognition/FaceRokognitionJs/MirarColecciones.js
require("dotenv").config();
const {
  RekognitionClient,
  ListCollectionsCommand,
} = require("@aws-sdk/client-rekognition");

//configura el cliente de Rekognition
const rekognitionClient = new RekognitionClient({
  region: process.env.AWS_REGION,
  credentials: {
    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
  },
});

async function listarColecciones() {
  try {
    const command = new ListCollectionsCommand({});
    const response = await rekognitionClient.send(command);
    console.log("Lista de colecciones:");
    response.CollectionIds.forEach((collectionId) => {
      console.log(collectionId);
    });
  } catch (error) {
    console.log("Error al listar las colecciones:", error);
  }
}

listarColecciones();

File: Fase3/aws-rekognition/FRpiton/MirarColecciones.py
import os
import boto3
from dotenv import load_dotenv
from botocore.exceptions import BotoCoreError, ClientError

# Cargar variables de entorno desde el archivo .env
load_dotenv()

# Configurar el cliente de Rekognition
rekognition_client = boto3.client(
    'rekognition',
    region_name=os.getenv("AWS_REGION"),
    aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
    aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY")
)

def listar_colecciones():
    try:
        response = rekognition_client.list_collections()
        print("Lista de colecciones:")
        for collection_id in response.get("CollectionIds", []):
            print(collection_id)
    except (BotoCoreError, ClientError) as error:
        print("Error al listar las colecciones:", error)

# Ejecutar la función
if __name__ == "__main__":
    listar_colecciones()
